Microsoft.AVS.VVOLS/Microsoft.AVS.VVOLS.psm1

```diff
@@ -0,0 +1,370 @@
+        # Unmount the datastore from the host
+        $datastoreSystem = Get-View -Id $Esxi.ExtensionData.ConfigManager.DatastoreSystem
+        Write-Host "Unmounting datastore $DatastoreName from host $($Esxi.Name)..."
+        $datastoreSystem.RemoveDatastore($Datastore.ExtensionData.MoRef)
+    }
+}
+
+
+<#
+    .SYNOPSIS
+    Refresh certificates for all ESXi hosts
+
+    .DESCRIPTION
+    Refresh certificates for all ESXi hosts
+
+    .EXAMPLE
+    Update-VMHostCertificate
+
+    .INPUTS
+    None.
+
+    .OUTPUTS
+    None.
+
+#>
+function Update-VMHostCertificate {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param ()
+
+    $service_instance = Get-View ServiceInstance
+    $certMgr = Get-View -Id $service_instance.Content.CertificateManager
+
+    Get-VMHost | ForEach-Object -Process {
+        Write-Host "Refreshing certificates for $($_.Name).."
+        $certMgr.CertMgrRefreshCACertificatesAndCRLs($_.Id) | Out-Null
+    }
+
+    Write-Host "Certificates refreshed successfully for all of ESXi hosts."
+}
+
+<#
+    .SYNOPSIS
+     Create a new VASA provider in vCenter
+
+    .PARAMETER ProviderName
+     Name of VASA provider to be created in vCenter
+
+    .PARAMETER FlashArrayMgmtIP
+     URL of the VASA provider service
+
+    .PARAMETER ProviderCredential
+     Credential of the VASA provider service
+
+    .EXAMPLE
+     New-VvolVasaProvider -ProviderName "myProvider" -FlashArrayMgmtIP "1.0.0.0" -ProviderCredential $ProviderCredential
+#>
+function New-VvolVasaProvider {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Name of VASA provider to be created in vCenter')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderName,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'URL of the VASA provider service')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderUrl,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Credential of the VASA provider service')]
+        [ValidateNotNull()]
+        [PSCredential]
+        $ProviderCredential
+    )
+
+    $VasaProvider = Get-VasaProvider -Name $ProviderName -ErrorAction Ignore
+    if ($vasaProvider) {
+        throw "Unable to create a VASA provider. Vasa provider '$ProviderName' already exists."
+    }
+
+    New-VasaProvider -Name $ProviderName -Credential $ProviderCredential -Url $ProviderUrl -Force -ErrorAction Stop | Out-Null
+
+    Write-Host "VASA provider $ProviderName created successfully."
+}
+
+<#
+    .SYNOPSIS
+     Remove a VASA provider from vCenter
+
+    .PARAMETER ProviderName
+     Name of VASA provider to be removed from vCenter
+
+    .EXAMPLE
+     Remove-VvolVasaProvider -ProviderName "myProvider"
+
+    .INPUTS
+     vCenter VASA provider name
+
+    .OUTPUTS
+     None.
+#>
+function Remove-VvolVasaProvider {
```

Comment:
Ditto - can we check that the provider is not in use. If not directly, then by looking at the datastores it provides?

Reply:
Done